components/mindstorms-ev3/src/main/java/com/example/ev3/EV3StatusAdapter.java

```diff
@@ -0,0 +1,765 @@
+    private boolean StreamPort3;
+    private boolean StreamPort4;
+
+    private int StreamPortAmode = 0;
+    private int StreamPortBmode = 0;
+    private int StreamPortCmode = 0;
+    private int StreamPortDmode = 0;
+    private int StreamPort1mode = 0;
+    private int StreamPort2mode = 0;
+    private int StreamPort3mode = 0;
+    private int StreamPort4mode = 0;
+
+    private SensorTypeEnum Port1Device;
+    private SensorTypeEnum Port2Device;
+    private SensorTypeEnum Port3Device;
+    private SensorTypeEnum Port4Device;
+
+    private boolean PortAMotor;
+    private boolean PortBMotor;
+    private boolean PortCMotor;
+    private boolean PortDMotor;
+
+    private EV3SharedObject connectTo;
+    public String ConnectionManagerName;
+
+    private String displayName = "MINDSTORMS EV3 Input Adapter";
+    // Local variables
+    private int inputPorts = 2;
+    private int outputPorts = 8;
+    private int nextOutputPort = 0;
+    private Schema[] outputSchemas; // caches the Schemas given during init() for use at processTuple()
+    private RobotPort[] botPortsInfo; // caches the schemas and port names being used
+    private HashMap<String, Integer> outputPortNames;
+
+    // Input schema definition
+    private static Schema.Field FIELD_TARGET_PORT = Schema.createField(DataType.STRING, "TargetPort");
+    private static Schema.Field FIELD_MODE = Schema.createField(DataType.STRING, "Mode");
+    private static Schema.Field FIELD_STREAM = Schema.createField(DataType.BOOL, "Stream");
+    // private static Schema.Field FIELD_SENSOR_MODE =
+    // Schema.createField(DataType.INT, "SensorMode");
+
+    // Output schema fieldnames
+    public static final String FIELD_DEGREES = "Degrees";
+    public static final String FIELD_ROTATION = "Rotation";
+    public static final String FIELD_POWER = "Power";
+
+    public static String FIELD_LEFT = "LeftButton";
+    public static String FIELD_CENTER = "CenterButton";
+    public static String FIELD_RIGHT = "RightButton";
+    public static String FIELD_UP = "UpButton";
+    public static String FIELD_DOWN = "DownButton";
+
+    public static final String FIELD_LISTEN = "Listen";
+    public static final String FIELD_TOUCH = "Touch";
+    public static final String FIELD_BUMPED = "Bumped";
+    public static final String FIELD_COLOR = "Color";
+    public static final String FIELD_REFLECT = "LightReflected";
+    public static final String FIELD_AMBIENT = "LightAmbient";
+    public static final String FIELD_DIST_CM = "DistanceCM";
+    public static final String FIELD_DIST_IN = "DistanceIN";
+    public static final String FIELD_ANGLE = "Angle";
+    public static final String FIELD_RATE = "Rate";
+    public static final String FIELD_PROXIMITY = "Proximity";
+    public static final String FIELD_REMOTE = "RemoteControl";
+
+    public EV3StatusAdapter() {
+        super();
+        setPortHints(inputPorts, outputPorts);
+
+        setDisplayName("MINDSTORMS Status Adapter for EV3");

```

Comment:
Could replace this with your displayName variable. If so, make it a final variable.